### PR TITLE
BHBC-1010: Modelling Permits to be associated with surveys in addition to projects

### PR DESCRIPTION
**Pull request by sdevalapurkar**
# Overview

## Links to jira tickets

- https://quartech.atlassian.net/browse/BHBC-1010

## This PR contains the following changes

- Modelling Permits to be associated with surveys in addition to projects
- App and API changes to accomodate single permit table (no sampling and sampling permits, ones associated and not associated to a project)
- Survey attachment table introduced

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [x] Enhancement (improvements to existing functionality)

## How Has This Been Tested?

- Locally, unit tests

### CODE REVIEW

File: api/src/paths/permit-no-sampling.ts [hunk 1]
```diff
@@ -104,7 +105,7 @@ export function createNoSamplePermits(): RequestHandler {
 
 export const insertNoSamplePermit = async (
   permit: IPostPermitNoSampling,
-  coordinator: PostCoordinatorData,
+  coordinator: PostCoordinatorData | PutCoordinatorData,
```
Comment (anissa-agahchen) [inline api/src/paths/permit-no-sampling.ts:108]: what'a happening here?

Comment (sdevalapurkar) [reply api/src/paths/permit-no-sampling.ts:108]: this function is now accepting a put or post object for the coordinator data cause we need to insert using this function from both the create and update project permit

File: api/src/models/project-update.ts [hunk 2]
```diff
@@ -344,7 +344,7 @@ export class GetPermitData {
           return {
             permit_number: item.number,
             permit_type: item.type,
-            sampling_conducted: item.sampling_conducted ? 'true' : 'false'
+            sampling_conducted: 'true'
```
Comment (anissa-agahchen) [inline api/src/models/project-update.ts]: can you leave a comment explaining this?

**Review by anissa-agahchen — APPROVED** [review]
Looks good

**Pull request merged**